### PR TITLE
core: Remove redundant NetCore classes and update client selection logic

### PR DESCRIPTION
**Pull request by cadatoiva**
The regular classes and the NetCore versions are Identical at this point, so reduce the WebClient classes from 4 -> 2.

The detect mono compatibility function was completely unused as well, so it's been removed in this version.

Tested working for both versions.

### CODE REVIEW

File: src/Jackett.Common/Plumbing/JackettModule.cs [hunk 1]
```diff
@@ -45,19 +43,14 @@ protected override void Load(ContainerBuilder builder)
...
                 case "httpclientnetcore":
-                    RegisterWebClient<HttpWebClientNetCore>(builder);
-                    break;
-                case "httpclient2netcore":
-                    RegisterWebClient<HttpWebClient2NetCore>(builder);
-                    break;
                 case "httpclient":
                     RegisterWebClient<HttpWebClient>(builder);
                     break;
+                case "httpclient2netcore":
```
Comment (ngosang) [inline src/Jackett.Common/Plumbing/JackettModule.cs]: The code is good but we can let a warning message with something like "the client httpclient2netcore was removed, using httpclient2 instead". Maybe some users are using ClientOverride parameter.

File: src/Jackett.Common/Utils/Clients/HttpWebClient2.cs [hunk 2]
```diff
@@ -67,6 +66,8 @@ public override void OnNext(ServerConfig value)
 
         public override void Init()
         {
+            ServicePointManager.DefaultConnectionLimit = 1000;
```
Comment (ngosang) [inline src/Jackett.Common/Utils/Clients/HttpWebClient2.cs]: This value is huge according to the default value 2 or 10. https://docs.microsoft.com/en-us/dotnet/api/system.net.servicepointmanager.defaultconnectionlimit?view=netframework-4.8
Maybe this is causing some network problems. Let this value as is in the PR, but we have to take a look in the future.

Comment (cadatoiva) [reply src/Jackett.Common/Utils/Clients/HttpWebClient2.cs]: I'll make a note of this. Currently this was in all other classes except HttpWebClient2, which is why it's been added to this one.

File: src/Jackett.Common/Plumbing/JackettModule.cs (hunk 1 above)
Comment (cadatoiva) [reply src/Jackett.Common/Plumbing/JackettModule.cs]: I'm going to be collapsing httpClient and httpClient2 in a future PR as well. At that point we can add a warning that the entire ClientOverride parameter is going to be removed as it doesn't do anything.

Comment (ngosang) [reply src/Jackett.Common/Plumbing/JackettModule.cs]: ok

**Comment by ngosang** [comment]
#8529

**Pull request merged**